File: RoadRunnerJS/src/jsMain/kotlin/trajectory/constraints/ProfileAccelerationConstraint.kt
package trajectory.constraints

import geometry.Pose2d
import kotlin.js.JsExport

/**
 * Constraint limiting profile acceleration.
 */
@JsExport
class ProfileAccelerationConstraint(
    private val maxProfileAccel: Double
) : TrajectoryAccelerationConstraint() {
    override fun get(s: Double, pose: Pose2d, deriv: Pose2d, baseRobotVel: Pose2d) = maxProfileAccel
}

File: src/roadrunnerjs/trajectory/constraints/ProfileAccelerationConstraint.ts
import { Pose2d } from "roadrunnerjs/geometry";
import { TrajectoryAccelerationConstraint } from "./TrajectoryAccelerationConstraint";
/**
 * Constraint limiting profile acceleration.
 */
export class ProfileAccelerationConstraint
  implements TrajectoryAccelerationConstraint
{
  constructor(private readonly maxProfileAccel: number) {}

  get(s: number, pose: Pose2d, deriv: Pose2d, baseRobotVel: Pose2d) {
    return this.maxProfileAccel;
  }
}
